Userland/Utilities/sort.cpp

```diff
@@ -109,16 +122,11 @@ ErrorOr<int> serenity_main([[maybe_unused]] Main::Arguments arguments)
     args_parser.add_positional_argument(options.files, "Files to sort", "file", Core::ArgsParser::Required::No);
     args_parser.parse(arguments);
 
-    Vector<Line> lines;
-    HashTable<Line> seen;
+    if (options.files.is_empty())
+        options.files.append("-"sv);
 
-    if (options.files.size() == 0) {
-        TRY(load_file(options, "-"sv, lines, seen));
-    } else {
-        for (auto& file : options.files) {
-            TRY(load_file(options, file, lines, seen));
-        }
-    }
+    auto file_list = TRY(read_files(options.files));
```

Comment:
In the old situation, `sort --unique ...` would go through the files one by one and deduplicate lines before continuing with the next file. In your version, you load all the input files to memory before parsing the lines. This can increase peak memory usage by a lot. Could you make your changes work with the old logic for handling the files?

Reply:
that might be another uniqueness algorithm, a custom extension with an option to choose, a global one (which might be of interest sometimes), i.e. what iʼm suggesting exactly is to keep the new algorithm with the old one to choose from (the old one being default).

Reply:
@konradekk Why would you do that? That would give you two "algorithms", both of which give the exact same result but one of them potentially uses much more memory than the other. The main difference here is the implementation, not the "algorithm" or complexity thereof.

Reply:
@tcl3 looking at this a bit further, you claim:

> My implementation of this feature avoids string allocations when building Line objects, reducing memory usage.

The old implementation would only keep unique lines around, your new implementation keeps a view to the entire file's contents right? Does that claim still hold for files with lots of duplicate lines?

Reply:
i understood that uniqueness is checked across all the provided files and this was the thing i was referring too. i was not looking exactly at the implementation—please ignore this dummy comment of mine. 🙇🏻‍♂️

Reply:
@GMTA I didn't fully consider the case where there would be a lot of duplicate lines when coming up with this solution. It was also misleading of me to say that my solution uses less memory than the original. It certainly leads to fewer memory allocations (which may be what I meant), but that isn't the same thing!

Rather than reading one file at a time, I thimk it would be better use a method similar to the original implementation, which reads a line at a time. This would be able to handle the case where you have a single huge file with lots of duplicate lines.

I think the best way forward is to implement a method to allow a stream to read an arbitrarily long line. There was a PR that implemented a `read_line_unbounded()` API. It looks like it got abandoned when the `Stream` code was moved. I can investigate the feasibility of reviving that, or I can implement something else that achieves the same goal.

Reply:
Now that we have the `read_until_with_resize()` method for reading arbitrarily long lines I've updated this PR to use that.